source/backend/api/Startup.cs

```diff
@@ -235,62 +237,109 @@ public void ConfigureServices(IServiceCollection services)
             // Export metrics from all HTTP clients registered in services
             services.UseHttpClientMetrics();
 
-            services.AddHealthChecks()
-                .AddCheck("liveliness", () => HealthCheckResult.Healthy())
-                .AddSqlServer(csBuilder.ConnectionString, tags: new[] { "services" })
-                .ForwardToPrometheus(); // Report health check results in the metrics output.
+            var allHealthCheckOptions = new AllHealthCheckOptions();
+            Configuration.GetSection("HealthChecks").Bind(allHealthCheckOptions);
+            services.AddHealthChecks().ForwardToPrometheus();
 
-            services.AddHealthChecks()
-                .AddCheck(
+            if (allHealthCheckOptions.SqlServer.Enabled)
+            {
+                services.AddHealthChecks()
+                    .Add(new HealthCheckRegistration(
+                        "SqlServer",
+                        sp =>
+                        {
+                            var options = new SqlServerHealthCheckOptions
+                            {
+                                ConnectionString = csBuilder.ConnectionString,
+                                CommandText = "SELECT 1;",
+                                Configure = null,
+                            };
+                            return new SqlServerHealthCheck(options);
+                        },
+                        default,
+                        new[] { "services" })
+                    { Period = TimeSpan.FromMinutes(allHealthCheckOptions.SqlServer.Period) });
+            }
+
+            if (allHealthCheckOptions.PimsDBCollation.Enabled)
+            {
+                services.AddHealthChecks().Add(new HealthCheckRegistration(
                     "PimsDBCollation",
-                    new PimsDatabaseHealthcheck(csBuilder.ConnectionString),
-                    HealthStatus.Unhealthy,
-                    new string[] { "services" });
+                    sp => new PimsDatabaseHealthcheck(csBuilder.ConnectionString),
+                    default,
+                    new string[] { "services" })
+                { Period = TimeSpan.FromMinutes(allHealthCheckOptions.PimsDBCollation.Period) });
+            }
 
-            services.AddHealthChecks()
-                .AddCheck(
-                    "api-metrics",
-                    new PimsMetricsHealthcheck(csBuilder.ConnectionString),
+            if (allHealthCheckOptions.ApiMetrics.Enabled)
+            {
+                services.AddHealthChecks().Add(new HealthCheckRegistration(
+                    "ApiMetrics",
```

Comment:
I believe we use the string "api-metrics" in our Prometheus/Sysdig dashboards. We will need to update the dashboards to read from the new "ApiMetrics" information or they will show no data...

Reply:
NVM - I checked Sysdig and we don't use that name at all - disregard previous comment.